frontend/lib/create_listing/view/create_listing.view.dart

```diff
@@ -0,0 +1,414 @@
+import 'package:utm_marketplace/shared/components/loading.component.dart';
+
+class CreateListingView extends StatefulWidget {
+  const CreateListingView({super.key});
+
+  @override
+  State<CreateListingView> createState() => _CreateListingViewState();
+}
+
+class _CreateListingViewState extends State<CreateListingView> {
+  final _formKey = GlobalKey<FormState>();
+  late CreateListingViewModel viewModel;
+  final _titleController = TextEditingController();
+  final _priceController = TextEditingController();
+  final _descriptionController = TextEditingController();
+  bool _showValidationErrors = false;
+
+  @override
+  void initState() {
+    super.initState();
+    viewModel = Provider.of<CreateListingViewModel>(context, listen: false);
+  }
+
+  // Header widgets
+  final appBarTitle = const Text(
+    'Create Post',
+    style: TextStyle(
+      fontSize: 32,
+      fontWeight: FontWeight.bold,
+    ),
+  );
+
+  final postButton = const Text(
+    'Post',
+    style: TextStyle(
+      color: Color(0xFF2F9CCF),
+      fontSize: 16,
+    ),
+  );
+
+  // Form widgets
+  Widget _buildImageSelector() {
+    return Consumer<CreateListingViewModel>(
+      builder: (_, model, __) {
+        final bool showError = !model.hasImage && _showValidationErrors;
+        
+        return GestureDetector(
+          onTap: () async {
+            // Your existing image selection logic
+          },
+          child: Container(
+            height: 300,
+            decoration: BoxDecoration(
+              color: Colors.grey[200],
+              borderRadius: BorderRadius.circular(12),
+              border: Border.all(
+                color: showError
+                    ? const Color(0xFFFF5252)
+                    : Colors.grey[300]!,
+                width: 1,
+              ),
+            ),
+            child: model.image != null
+                ? ClipRRect(
+                    borderRadius: BorderRadius.circular(12),
+                    child: Image.file(
+                      model.image!,
+                      fit: BoxFit.cover,
+                      width: double.infinity,
+                    ),
+                  )
+                : Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      Icon(
+                        Icons.add_photo_alternate_outlined,
+                        size: 48,
+                        color: showError
+                            ? const Color(0xFFFF5252)
+                            : Colors.grey[600],
+                      ),
+                      const SizedBox(height: 8),
+                      Text(
+                        'Add photos / video',
+                        style: TextStyle(
+                          color: showError
+                              ? const Color(0xFFFF5252)
+                              : Colors.grey[600],
```

Comment:
As usual, please split this up into smaller sub-widgets to maintain readability.